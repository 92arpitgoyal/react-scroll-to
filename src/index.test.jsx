import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import { ScrollTo, ScrollToHOC } from "./";

beforeEach(() => {
  window.scroll = jest.fn();
});

describe("Test ScrollTo Component.", () => {
  describe("Test render prop.", () => {
    it("Should render the functional children.", () => {
      const wrapper = shallow(<ScrollTo>{scroll => <div>test</div>}</ScrollTo>);

      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it("Should render nothing nothing on undefined.", () => {
      const wrapper = shallow(<ScrollTo>{undefined}</ScrollTo>);

      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it("Should call window.scroll.", () => {
      const wrapper = shallow(
        <ScrollTo>
          {scroll => <button onClick={() => scroll(100, 200)}>test</button>}
        </ScrollTo>
      );

      const buttonEl = wrapper.find("button");
      buttonEl.simulate("click");

      expect(window.scroll).toHaveBeenCalledTimes(1);
      expect(window.scroll.mock.calls[0]).toEqual([100, 200]);
    });
  });

  describe("Test HOC.", () => {
    it("Should render the functional children.", () => {
      const TestComponent = () => <div>test</div>;
      TestComponent.displayName = "test";
      const WrappedComponent = ScrollToHOC(TestComponent);
      const wrapper = mount(<WrappedComponent />);

      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it("Should call window.scroll.", () => {
      const TestComponent = props => (
        <button onClick={() => props.scroll(100, 200)}>test</button>
      );
      TestComponent.displayName = "test";
      const WrappedComponent = ScrollToHOC(TestComponent);
      const wrapper = mount(<WrappedComponent />);

      const buttonEl = wrapper.find("button");
      buttonEl.simulate("click");

      expect(window.scroll).toHaveBeenCalledTimes(1);
      expect(window.scroll.mock.calls[0]).toEqual([100, 200]);
    });
  });
});
