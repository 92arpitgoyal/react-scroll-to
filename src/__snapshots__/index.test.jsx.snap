// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ScrollTo Component. Test HOC. Should render the functional children. 1`] = `
<WithScrollToHOC(test)>
  <test
    scroll={[Function]}
  >
    <div>
      test
    </div>
  </test>
</WithScrollToHOC(test)>
`;

exports[`Test ScrollTo Component. Test render prop. Should render nothing nothing on undefined. 1`] = `""`;

exports[`Test ScrollTo Component. Test render prop. Should render the functional children. 1`] = `
<div>
  test
</div>
`;
